tests/e2e/e2e_util_test.go

```diff
@@ -863,3 +878,103 @@ func (s *IntegrationTestSuite) executeGaiaKeysAddCommand(ctx context.Context, c
 
 	return addrRecord.Address
 }
+
+func (s *IntegrationTestSuite) executeKeysList(c *chain, valIdx int, home string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	gaiaCommand := []string{
```

Comment:
This kind of operation I've isolated in a helper fn, seems unnecessary to repeat

Reply:
Yeah so there's the main helper function `executeGaiaTxCommand` and the difference in this case is it queries the transaction hash after executing to confirm the transaction didn't fail...we could refactor this further to a general purpose function that know whether it's executing a tx or keys command but I fear refactoring too much at this point, lmk what you think if we should go ahead and just do this